### PR TITLE
Manual page reference injection option added.

### PR DESCRIPTION
**Pull request by mfendeksilverstripe**
Manual page reference injection option added to support cases where Page_Controller is not available.

### CODE REVIEW

File: src/parsers/AccordionParser.php [hunk 1]
```diff
@@ -23,14 +30,44 @@ public static function get_shortcodes()
...
         if (Controller::curr() instanceof Page_Controller) {
-            $accordionItems = Controller::curr()->dataRecord;
-            $template = 'AccordionItems';
-            $ssViewer = new SSViewer($template);
+            // Only if we're on a Page, so the CMS doesn't crash.
+            $page = Controller::curr()->dataRecord;
+        }
+        // attempt to load page via custom loader
+        else {
+            $page = self::$page;
```
Comment (Firesphere) [inline src/parsers/AccordionParser.php]: Preferably use `static::$page`

Comment (mfendeksilverstripe) [reply src/parsers/AccordionParser.php]: It's fixed now.

File: src/parsers/AccordionParser.php [hunk 2]
```diff
@@ -23,14 +30,44 @@ public static function get_shortcodes()
...
+
+    /**
+     * Use this in the custom parsing case (non Page_Controller case)
+     * this is handy in cases where page can't be injected via controller
+     * page has to support accordions
+     * @param Page $page
+     */
+    public static function setPage(Page $page)
+    {
+        self::$page = $page;
```
Comment (Firesphere) [inline src/parsers/AccordionParser.php]: These need to be `static` as well ;)

Comment (mfendeksilverstripe) [reply src/parsers/AccordionParser.php]: Done :)

**Pull request merged**